src/libraries/Common/src/System/Net/Http/Http3/QPack/H3StaticTable.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+#if KESTREL
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack
+#else
+namespace System.Net.Http.QPack
+#endif
+{
+    internal class H3StaticTable
+    {
+        private static readonly H3StaticTable _instance = new H3StaticTable();
+
+        private readonly Dictionary<int, int> _statusIndex = new Dictionary<int, int>
+        {
+            [103] = 24,
+            [200] = 25,
+            [304] = 26,
+            [404] = 27,
+            [503] = 28,
+            [100] = 63,
+            [204] = 64,
+            [206] = 65,
+            [302] = 66,
+            [400] = 67,
+            [403] = 68,
+            [421] = 69,
+            [425] = 70,
+            [500] = 71,
+        };
+
+        private readonly Dictionary<HttpMethod, int> _methodIndex = new Dictionary<HttpMethod, int>
+        {
+            // TODO connect is internal to system.net.http
+            [HttpMethod.Delete] = 16,
+            [HttpMethod.Get] = 17,
+            [HttpMethod.Head] = 18,
+            [HttpMethod.Options] = 19,
+            [HttpMethod.Post] = 20,
+            [HttpMethod.Put] = 21,
+        };
+
+        private H3StaticTable()
+        {
+        }
+
+        public static H3StaticTable Instance => _instance;
+
+        public int Count => _staticTable.Length;
+
+        public HeaderField this[int index] => _staticTable[index];
+
+        public IReadOnlyDictionary<int, int> StatusIndex => _statusIndex;
+        public IReadOnlyDictionary<HttpMethod, int> MethodIndex => _methodIndex;
```

Comment:
Why not just expose this as `Dictionary<>` rather than `IReadOnlyDictionary<>`? I get that the latter helps us avoid accidental writes from other code internal to the app, but it also forces all use to go through interface dispatch.